integration/business-process-tests/tests-integration/bpmn/src/test/java/org/wso2/ei/businessprocess/integration/tests/bpmn/BPMNPackageDeploymentResolveTestCase.java

```diff
@@ -0,0 +1,110 @@
+        super.init();
+        serverConfigurationManager = new ServerConfigurationManager(bpsServer);
+        updateConfigFiles();
+        super.init();
+        startupParameterMap.put("-DportOffset", "0");
+        startupParameterMap.put("-DresolveDeploymentsAtStartup", "false");
+        startupParameterMap.put("startupScript", "business-process");
+        server = new CarbonTestServerManager(bpsServer, System.getProperty("carbon.zip"), startupParameterMap);
+        server.startServer();
+        loginLogoutClient.login();
+        workflowServiceClient = new WorkflowServiceClient(backEndUrl, sessionCookie);
+        serverConfigurationManager = new ServerConfigurationManager(bpsServer);
+    }
+
+    private void updateConfigFiles() throws Exception {
+        final String artifactLocation =
+                FrameworkPathUtil.getSystemResourceLocation() + BPMNTestConstants.DIR_ARTIFACTS + File.separator
+                        + BPMNTestConstants.DIR_CONFIG + File.separator;
+        //Adding new config file for activiti.xml
+        File activitiConfigNew = new File(artifactLocation + BPMNTestConstants.ACTIVITI_CONFIGURATION_FILE_NAME);
+        File activitiConfigOriginal = new File(
+                FrameworkPathUtil.getCarbonHome() + File.separator + NEW_CONF_DIR + File.separator
+                        + BPMNTestConstants.ACTIVITI_CONFIGURATION_FILE_NAME);
+        serverConfigurationManager.applyConfiguration(activitiConfigNew, activitiConfigOriginal, true, true);
+    }
+
+    @Test(description = "test bpmn artifact undeploying with startup server resolve parameter",
+            priority = 1, singleThreaded = true)
+    public void testBPMNUndeployWithServerStartupResolving() throws Exception {
+        uploadBPMNForTest(BPMN_PACKAGE_NAME);
+        BPMNTestUtils.waitForProcessDeployment(workflowServiceClient, BPMN_PACKAGE_NAME, 0);
+        Assert.assertTrue("Error occurred while deploying bpmn artifact",
+                workflowServiceClient.getDeployments().length > 0);
+        Files.deleteIfExists(Paths.get(FrameworkPathUtil.getCarbonHome() + File.separator + "wso2" + File.separator +
+                "business-process" + File.separator + "repository" + File.separator + "deployment" + File.separator +
+                "server" + File.separator + "bpmn" + File.separator + "HelloApprove.bar"));
+        BPMNTestUtils.waitForProcessDeployment(workflowServiceClient, BPMN_PACKAGE_NAME, 0);
+        //Check whether the deployed artifacts are undeployed
+        Assert.assertNotNull(workflowServiceClient.getDeployments());
+        Assert.assertNotNull(workflowServiceClient.getProcesses());
+    }
+
+    @Test(description = "test bpmn artifact undeploying without startup server resolve parameter",
+            priority = 2, singleThreaded = true)
+    public void testBPMNUndeployWithoutServerStartupResolving() throws Exception {
+        workflowServiceClient.undeploy(BPMN_PACKAGE_NAME);
+        server.stopServer();
```

Comment:
please make sure to stop server in the after class as well, if the test method skipped, other test will run again the server started in this test case.